File: sdk/node/src/api/course/course.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class CourseCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  
    @IsString()
    TenantId: string;
    @IsString()
    Name: string;
    @IsString()
    Description: string;
    @IsString()
    ImageUrl: string;
    @IsNumber()
    DurationInDays: number;

}

export class CourseUpdateModel {
    @IsOptional()
    @IsString()
    TenantId?: string;
    @IsOptional()
    @IsString()
    Name?: string;
    @IsOptional()
    @IsString()
    Description?: string;
    @IsOptional()
    @IsString()
    ImageUrl?: string;
    @IsOptional()
    @IsNumber()
    DurationInDays?: number;

}

export class CourseSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}

File: sdk/dotnet/lmsservice_db/src/api/course/CourseModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.course.coursemodel;
public class CourseCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;


}

public class CourseUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;


}

public class CourseSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

